File: Debugging/Gdb/hello_world.f90
program hello_world
     
  integer*4 nthreads, tid, &
       &    OMP_GET_NUM_THREADS, OMP_GET_THREAD_NUM
  
  ! Fork a team of threads giving them their own copies of variables
  !$OMP PARALLEL PRIVATE(nthreads, tid)

  ! Obtain thread number
  tid = OMP_GET_THREAD_NUM()
  print *, 'Hello World from thread = ', tid

  ! Only master thread does this
  if (tid == 0) then
     nthreads = OMP_GET_NUM_THREADS()
     print *, 'Number of threads = ', nthreads
  end if

  ! All threads join master thread and disband
  !$OMP END PARALLEL

end program hello_world

File: Debugging/Gdb/C/hello_world.c
#include <omp.h>
#include <stdio.h>

int main() {
#pragma omp parallel
    {
        int thread_num = omp_get_thread_num();
        printf("thread %d\n", thread_num);
        if (thread_num == 0) {
            int num_threads = omp_get_num_threads();
            printf("%d total threads\n", num_threads);
        }
    }
    return 0;
}
